onnxruntime/core/optimizer/attention_fusion.cc

```diff
@@ -280,8 +281,7 @@ Status AttentionFusion::ApplyImpl(Graph& graph, bool& modified, int graph_level,
           reshape_count++;
         }
       }
-
-      if (add_count == 1 && matmul_count == 3) { // BERT
+      if (add_count == 1 && matmul_count == 3 && (shape_count >= 0 && shape_count <= 2)) {  // BERT or DistilBert
```

Comment:
change shape_count >= 0 && shape_count <= 2 to shape_count == node.GetOutputEdgesCount() - 4